### PR TITLE
test: add tests to `stats/base/dists/geometric/mgf`

### PR DESCRIPTION
**Pull request by whizzel**
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: na
  - task: lint_package_json status: na
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: na
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

Resolves None .

## Description

> What is the purpose of this pull request?

This pull request:

-   **Current Coverage:**
- Statements: 284/288 (98.6%)
- Branches: 17/19 (89.5%)
- Functions: 4/4 (100%)

**Missing Coverage Areas:**
1. **Boundary condition**: `t = -ln(1-p)` → should return `NaN`
2. **Just below boundary**: `t < -ln(1-p)` → should return finite value
3. **Just above boundary**: `t > -ln(1-p)` → should return `NaN`
4. **Edge case**: `p = 0` → should return `NaN`
5. **Edge case**: `p = 1` → should return `e^t`
6. **Very small p**: boundary condition handling
7. **Factory pattern**: boundary conditions
 
**Impact:**
- Achieve 100% test coverage
- Ensure numerical stability edge cases are tested
- Improve reliability of statistical computations
- This is a pure testing improvement that enhances reliability without affecting the API.

## Related Issues

> Does this pull request have any related issues?

This pull request has the following related issues:

-   #{{TODO: add related issue number}}

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

> When authoring the changes proposed in this PR, did you use any kind of AI assistance?

-   [x] Yes
-   [ ] No

If you answered "yes" above, how did you use AI assistance?

-   [ ] Code generation (e.g., when writing an implementation or fixing a bug)
-   [ ] Test/benchmark generation
-   [ ] Documentation (including examples)
-   [x] Research and understanding

#### Disclosure

> If you answered "yes" to using AI assistance, please provide a short disclosure indicating how you used AI assistance. This helps reviewers determine how much scrutiny to apply when reviewing your contribution. Example disclosures: "This PR was written primarily by Claude Code." or "I consulted ChatGPT to understand the codebase, but the proposed changes were fully authored manually by myself.".

{{TODO: add disclosure if applicable}}

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by whizzel** [comment]
Hello @kgryte,

Following our previous discussion in https://github.com/stdlib-js/stdlib/pull/10361#issuecomment-3948200078, I've submitted a new PR focused on improving test coverage for the geometric MGF module (stats/base/dists/geometric/mgf). 


if this is okey i would love to fix in others modules too.

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js [hunk 1]
```diff
@@ -118,3 +121,64 @@ tape( 'the function evaluates the mgf for `x` given large parameter `p`', functi
...
+
+tape( 'the function returns NaN when t equals the boundary condition t = -ln(1-p)', function test( t ) {
+	var boundary;
+	var p;
+	var y;
+
+	p = 0.5;
+	boundary = -ln( 1 - p );
+	y = mgf( boundary, p );
+	t.strictEqual( isnan( y ), true, 'returns NaN at exact boundary' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js]: Assertion messages throughout the new tests should be `'returns expected value'` to match the convention used in this file and across the project. Same applies to all new assertions in `test.factory.js` and `test.js`.

```suggestion
	t.strictEqual( isnan( y ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.js [hunk 2]
```diff
@@ -36,3 +38,15 @@ tape( 'attached to the main export is a factory method for generating `mgf` func
...
+	var boundary;
+	var p;
+	var y;
+
+	p = 0.5;
+	boundary = -ln( 1 - p );
+	y = mgf( boundary, p );
+	t.strictEqual( isnan( y ), true, 'returns NaN at boundary condition' );
+	t.end();
+});
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.js]: `test.js` is reserved for basic export checks (e.g., "is a function", "has factory method"). Mathematical behavior tests belong in `test.mgf.js`. This test and the added imports should be removed from this file.

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.factory.js [hunk 3]
```diff
@@ -148,3 +149,15 @@ tape( 'the created function evaluates the mgf for `x` given large parameter `p`'
...
+
+tape( 'the factory function returns NaN when t equals the boundary condition t = -ln(1-p)', function test( t ) {
+	var boundary;
+	var mgf;
+	var y;
+
+	mgf = factory( 0.5 );
+	boundary = -ln( 1 - 0.5 );
+	y = mgf( boundary );
+	t.strictEqual( isnan( y ), true, 'factory returns NaN at boundary' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.factory.js]: assertion message below needs updating:

```suggestion
	t.strictEqual( isnan( y ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js [hunk 4]
```diff
@@ -118,3 +121,64 @@ tape( 'the function evaluates the mgf for `x` given large parameter `p`', functi
 	}
 	t.end();
 });
+
+tape( 'the function returns NaN when t equals the boundary condition t = -ln(1-p)', function test( t ) {
```
Comment (Neerajpathak07) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js:125]: Any reason why these test cases were not added for the native version.

**Review by Neerajpathak07 — CHANGES_REQUESTED** [review]
@ivishal-g Thanks for working on this.

**Comment by whizzel** [comment]
Hello @Neerajpathak07, 

Thanks for pointing this out. I’ve added the corresponding tests for the native implementation in the latest commit. Please let me know if this looks good, and I’ll follow the same pattern in other related modules which do not yet have 100% test-cov .

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Left some feedback.

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.native.js [hunk 5]
```diff
@@ -44,6 +41,12 @@ var opts = {
 };
 
 
+// FIXTURES //
+
+var smallP = require( './fixtures/julia/small_p.json' );
+var largeP = require( './fixtures/julia/large_p.json' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.native.js]: FIXTURES should come before VARIABLES.

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.native.js [hunk 6]
```diff
@@ -92,18 +95,20 @@ tape( 'the function evaluates the mgf for `x` given small parameter `p`', opts,
 	p = smallP.p;
 	for ( i = 0; i < x.length; i++ ) {
 		y = mgf( x[i], p[i] );
-		if ( y === expected[i] ) {
-			t.strictEqual( y, expected[i], 'x: '+x[i]+', p: '+p[i]+', y: '+y+', expected: '+expected[i] );
-		} else {
-			delta = abs( y - expected[ i ] );
-			tol = 3.0 * EPS * abs( expected[ i ] );
-			t.ok( delta <= tol, 'within tolerance. x: '+x[ i ]+'. p: '+p[i]+'. y: '+y+'. E: '+expected[ i ]+'. Δ: '+delta+'. tol: '+tol+'.' );
+		if ( expected[i] !== null ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.native.js]: The fixture data doesn't contain any `null` values, so this guard is unnecessary. It also changes existing test logic without justification and creates an inconsistency with `test.mgf.js`, which doesn't have this guard. Please revert the fixture loop logic to match the original.

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js [hunk 7]
```diff
@@ -118,3 +121,64 @@ tape( 'the function evaluates the mgf for `x` given large parameter `p`', functi
...
+	y = mgf( boundary + 1e-15, p );
+	t.strictEqual( isnan( y ), true, 'MGF is not defined when t is just above boundary condition' );
+	t.end();
+});
+
+tape( 'the function returns NaN when p equals 0', function test( t ) {
+	var y;
+
+	y = mgf( 0.5, 0.0 );
+	t.strictEqual( isnan( y ), true, 'returns NaN for p = 0' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js]: Assertion messages should consistently use `'returns expected value'` per stdlib convention. The custom messages like `'returns NaN for p = 0'`, `'returns e^t for p = 1'`, `'MGF is not defined when t is just above boundary condition'`, etc., deviate from the standard pattern used throughout the existing tests.

```suggestion
	t.strictEqual( isnan( y ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js [hunk 8]
```diff
@@ -118,3 +121,64 @@ tape( 'the function evaluates the mgf for `x` given large parameter `p`', functi
...
+
+tape( 'the function returns NaN when t is just above the boundary condition', function test( t ) {
+	var boundary;
+	var p;
+	var y;
+
+	p = 0.5;
+	boundary = -ln( 1 - p );
+	y = mgf( boundary + 1e-15, p );
+	t.strictEqual( isnan( y ), true, 'MGF is not defined when t is just above boundary condition' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js]: Same here — use `'returns expected value'` for the assertion message.

```suggestion
	t.strictEqual( isnan( y ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js [hunk 9]
```diff
@@ -118,3 +121,64 @@ tape( 'the function evaluates the mgf for `x` given large parameter `p`', functi
...
+	y = mgf( 0.5, 0.0 );
+	t.strictEqual( isnan( y ), true, 'returns NaN for p = 0' );
+	t.end();
+});
+
+tape( 'the function returns e^t when p equals 1', function test( t ) {
+	var y;
+
+	y = mgf( 0.5, 1.0 );
+	t.strictEqual( y, exp( 0.5 ), 'returns e^t for p = 1' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js]:
```suggestion
	t.strictEqual( y, exp( 0.5 ), 'returns expected value' );
```

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js [hunk 10]
```diff
@@ -118,3 +121,64 @@ tape( 'the function evaluates the mgf for `x` given large parameter `p`', functi
...
+	t.end();
+});
+
+tape( 'the function returns NaN for very small p values due to boundary condition', function test( t ) {
+	var y;
+
+	y = mgf( 0.001, 1e-10 );
+	t.strictEqual( isnan( y ), true, 'returns NaN for very small p due to boundary condition' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.mgf.js]:
```suggestion
	t.strictEqual( isnan( y ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.native.js [hunk 11]
```diff
@@ -117,13 +122,76 @@ tape( 'the function evaluates the mgf for `x` given large parameter `p`', opts,
...
+	y = mgf( boundary + 1e-15, p );
+	t.strictEqual( isnan( y ), true, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function returns NaN when p equals 0', opts, function test( t ) {
+	var y;
+
+	y = mgf( 0.5, 0.0 );
+	t.strictEqual( isnan( y ), true, 'returns NaN for p = 0' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.native.js]: Same assertion message issue as in `test.mgf.js` — use `'returns expected value'` throughout for consistency.

```suggestion
	t.strictEqual( isnan( y ), true, 'returns expected value' );
```

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.native.js [hunk 12]
```diff
@@ -117,13 +122,76 @@ tape( 'the function evaluates the mgf for `x` given large parameter `p`', opts,
...
+	t.end();
+});
+
+tape( 'the function returns e^t when p equals 1', opts, function test( t ) {
+	var y;
+
+	y = mgf( 0.5, 1.0 );
+	t.strictEqual( y, exp( 0.5 ), 'returns e^t for p = 1' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.native.js]:
```suggestion
	t.strictEqual( y, exp( 0.5 ), 'returns expected value' );
```

File: lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.native.js [hunk 13]
```diff
@@ -117,13 +122,76 @@ tape( 'the function evaluates the mgf for `x` given large parameter `p`', opts,
...
+	t.end();
+});
+
+tape( 'the function returns NaN for very small p values due to boundary condition', opts, function test( t ) {
+	var y;
+
+	y = mgf( 0.001, 1e-10 );
+	t.strictEqual( isnan( y ), true, 'returns NaN for very small p due to boundary condition' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/geometric/mgf/test/test.native.js]:
```suggestion
	t.strictEqual( isnan( y ), true, 'returns expected value' );
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]

**Review by Planeshifter — APPROVED** [review]
Thanks!
